[docu-2785] Custom reports (#5098)
Kong/docs.konghq.com: app/konnect/analytics/custom-reports/use-cases.md

---

> title: Use Cases

Reviewer: ```suggestion
title: Report Use Cases
```
I think "Use Cases" would probably be fine in the nav bar, but I think expanding the title here would help for searches on Google/help users who come here from a search.

content_type: tutorial
badge: plus
---

Custom reporting in {{site.konnect_saas}} gives you the power to monitor your API data in detail and export that data to CSV. 

> This tutorial guides you through the process of creating custom reports for key API metrics using {{site.konnect_saas}} custom reports. 

Reviewer: ```suggestion
This tutorial guides you through determining which custom report to create based on KPIs and the process of creating those custom reports for key API metrics. 
```
Not sure if I love the wording I suggested, I just wanted to get the point across that this tutorial both helps you figure out which report to create based on your KPIs and then we also guide you through how to create said reports.

> You have just joined an organization as an API product manager. Your first task is to create a few reports that model business KPIs so that the executive team and investors have a grasp on the state of the API. You've selected the following KPIs because they would give a brief but wholistic understanding of the API: 

Reviewer: ```suggestion
In this tutorial scenario, you just joined an organization as an API product manager. Your first task is to create a few reports that model business KPIs so that the executive team and investors have a grasp on the state of the API. You've selected the following KPIs because they would give a brief, but wholistic, understanding of the API: 
```

* **Number of requests:** This metric measures the total number of requests made to the API. This is good for understanding total usage of an API, and can potentially be used to uncover scalability requirements. 

> * **Error response rate:** This metric measures the percentage of requests that result in errors. With this metric you can potentially pinpoint any issues users of your API are having. 

Reviewer: ```suggestion
* **Error response rate:** This metric measures the percentage of requests that result in errors. With this metric, you can potentially pinpoint any issues users of your API are having. 
```

* **Average throughput:** Throughput is the measurement of the number of requests an API can handle per second. 

> * **Latency:** latency is the measurement of the time it takes for the API to respond to a request. 

Reviewer: ```suggestion
* **Latency:** This metric measures the time it takes for the API to respond to a request. This can be a good measure of the user experience  when a user makes a request to an API.
```

### Latency for an API over the last 30 days

At this point, you have multiple reports that display how your company's APIs are performing generally.

Your company determines that it is critical that payments are processed quickly using the Payments API, so you decide to generate a report that will tell you how performant the Payment API is over the last 30 days. To build the report, set the following options in the UI:

* **Name**: Payment API - Latency (last 30 days)
* **Date/Time**: Last 30 days
* **Chart type**: Line chart
* **Select a metric**: Response latency (p99)
* **Group by**: Service
* **Choose granularity**: Daily
* **Entity Selection**: Payment 

Add a filter for the Payment API. Click on **Add Filter**, then set the following options:
 
* **Filter by**: Service
* **Operator**: In
* **Value**: Payment

[screenshot of finished report]

This report can provide your stakeholders with the answers to questions like:

* How performant has my API been over the last 30 days?
* Is my Payment API processing payments quickly enough?
* Are there any performance outliers?

> ## Conclusion 

Reviewer: I wanted to relate this back to the KPIs we listed earlier, but wasn't sure if this was the best way or if the conclusion should contain something else?

* If there are anomalies, do they repeat? Are there any patterns in the data?
* If there's an issue with API traffic, has it been resolved or is it ongoing?

## Daily API usage by application over the last 30 days

At this point you have two custom reports: 

* Total number of requests across APIs. 
* Daily API Usage for an API over the last 30 days.

These two custom reports can work together to provide you with a lot of information about your organizations product. These alone can help drive a lot internal change to your API. Let's dive deeper, if you want to learn about the types of third-party applications that your users use to connect to your API, set the following options in the UI: 

* **Name**: API Usage by Application (last 30 days)
* **Chart type**: Vertical bar chart
* **Date/Time**: Last 30 days
* **Metric**: Request count
* **Group by**: Service
* **Then by**: Application

[screenshot of finished report]

This report can be used to highlight which applications users prefer, where to distribute resources, and combined with the other reports, you can now communicate usage data about your API to your stakeholders. 
## Latency for an API over the last 30 days

> This use case demonstrates how you can use custom reports to track the latency for an API, the Payment API in this case, over the last 30 days.

Reviewer: @Guaris @lena-larionova I'm not sure if I should keep these as steps or not. I like the format of steps, but I don't think a user could follow this from start to finish because I never explain how to set up the "Payment" API. I feel like that's a bit outside of the scope of this section too. 

I think I could either add in info about how to create the Payment API and leave the steps, or match the sections you wrote where we don't have steps and it's more casual of "here's how you'd configure this" vs. "follow this from step 1 to step 2". Thoughts?

Reviewer 2: > or match the sections you wrote where we don't have steps and it's more casual of "here's how you'd configure this" vs. "follow this from step 1 to step 2". Thoughts?

I think this is the way to go. I don't think creating/configuring a service is in scope here. 

What we can do is introduce the APIs in the narrative. You're this person, and you have these APIs that you're responsible for: Payments, Accounts, and <something>.

Author: I agree with Lena.

## Daily API usage for an API over the last 30 days

From the previous report, you determine that the Accounts API is receiving the most traffic. 
You don't know whether this is a cause for concern or not, so you decide to take a closer look.

To gain further insight into the Accounts API, let's build a report that displays the number of requests per day for the Accounts API over the last 30 days. Set the following options in the UI:

* **Name**: Daily Accounts API Usage (last 30 days)
* **Chart type**: Line chart
* **Date/Time**: Last 30 days
* **Metric**: Request count
* **Group by**: Service

Add a filter for the Account API. Click on **Add Filter**, then set the following options:

* **Filter by**: Service
* **Filter value**: Accounts API (or your own API name)

[screenshot of finished report]

This report can provide your stakeholders with the answers to questions like:

* What does average traffic over time look like for this service?
* Are there any anomalies, such as spikes or drops in traffic?
* If there are anomalies, do they repeat? Are there any patterns in the data?
* If there's an issue with API traffic, has it been resolved or is it ongoing?

> <<<<<<< HEAD

Reviewer: ```suggestion
```

> ## Latency for an API over the last 30 days
> 
> This use case demonstrates how you can use custom reports to track the latency for an API, the Payment API in this case, over the last 30 days.

Reviewer: ```suggestion
```

> =======

Reviewer: ```suggestion
```

## Daily API usage by application over the last 30 days

At this point you have two custom reports: 

* Total number of requests across APIs. 
* Daily API Usage for an API over the last 30 days.

These two custom reports can work together to provide you with a lot of information about your organizations product. These alone can help drive a lot internal change to your API. Let's dive deeper, if you want to learn about the types of third-party applications that your users use to connect to your API, set the following options in the UI: 

* **Name**: API Usage by Application (last 30 days)
* **Chart type**: Vertical bar chart
* **Date/Time**: Last 30 days
* **Metric**: Request count
* **Group by**: Service
* **Then by**: Application

[screenshot of finished report]

This report can be used to highlight which applications users prefer, where to distribute resources, and combined with the other reports, you can now communicate usage data about your API to your stakeholders. 

> ## Latency for the Payment API over the last 30 days
> >>>>>>> 40e9d9d6297126072c1941cdbcc4ceb7dd6a1614

Reviewer: ```suggestion
## Latency for an API over the last 30 days

This use case demonstrates how you can use custom reports to track the latency for an API, the Payment API in this case, over the last 30 days.
```
